### PR TITLE
Week2 datastore

### PR DESCRIPTION
**Pull request by BenMob**
# Comment Section Added

### Deployed Version--> [https://bombeni1-sps-summer20.uc.r.appspot.com](url)

Note: For some reason, this link is returning a `404 error `if you just click on it, however it works just fine if  pasted in the browser.

Added a few new classes,

1. `com.google.sps.entities.Comment` : The Comment entity
2. `com.google.sps.services.Services`: Contains static helper methods that perform tasks such as converting a java Object into JSON.
3. `com.google.sps.services.commentServices`: **extends**  `com.google.sps.services.Services` and contains methods that perform tasks that are specific for **Comments** such as saving and retrieviing comments to and from datastore.

4. `com.google.sps.servlets.CommentServlet`: As the name says,  receives requests from the clients and uses the classes above to process them.

Thank you! :)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/services/CommentService.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+/**********************************************
+ * This class contains all the service methods
+ * to process a Comment Object.
+ *
+ * Child class to: com.google.sps.services.CommentServices
+ *
+ * TODO: Consider doing exeption handling where appropriate
+ */
+ import com.google.appengine.api.datastore.Entity;
```
Comment (gitmnk) [inline portfolio/src/main/java/com/google/sps/services/CommentService.java]: alphabetical order.

**Review by gitmnk — CHANGES_REQUESTED** [review]
Looks great!

File: portfolio/src/main/java/com/google/sps/services/Services.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ public class Services{
+
+   /**********************************************
+    * Converts a Java Object into JSON String
+    * @param : A Java Object
+    * @return : Json String 
+    **/
+    public static String toJson(Object o){
+        Gson gson = new Gson();
+        return gson.toJson(o);
```
Comment (qidichen) [inline portfolio/src/main/java/com/google/sps/services/Services.java]: If you want to be more concise, 
```suggestion
        return new Gson().toJson(o);
```

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 3]
```diff
@@ -19,14 +19,31 @@
...
-@WebServlet("/data")
-public class DataServlet extends HttpServlet {
-
+@WebServlet("/comments")
+public class CommentServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    ArrayList<Comment> comments = CommentService.getAllComments();
```
Comment (qidichen) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: Better to use interface 
```suggestion
    List<Comment> comments = CommentService.getAllComments();
```

File: portfolio/src/main/java/com/google/sps/services/CommentService.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+          datastore.put(commentEntity);
+      }
+
+       /********************************************************************
+      * Queries all the Comments from datatsore in ascending order based on
+      * timestamps.
+      *
+      * @return: java.util.ArrayList<Comment> 
+      */
+      public static ArrayList<Comment> getAllComments(){
```
Comment (qidichen) [inline portfolio/src/main/java/com/google/sps/services/CommentService.java]: Better to use interface List<...>

**Review by qidichen — APPROVED** [review]
Looks good!

File: portfolio/src/main/java/com/google/sps/services/CommentService.java (hunk 4 above)
Comment (BenMob) [reply portfolio/src/main/java/com/google/sps/services/CommentService.java]: I see.

**Pull request merged**